src/core/markdown.js

```diff
@@ -73,13 +73,13 @@ class Renderer extends marked.Renderer {
     return { language, ...metaData };
   }
 
-  heading(text, level, raw, slugger) {
+  heading(text, level, raw) {
```

Comment:
@sidvishnoi, will the slugger matter for id generation for the custom header syntax? 

It seems marked moved the slugger thing to an independent module for adding ids to headers.

Reply:
Maybe it won't matter because no test failed? (presumably we test our custom syntax)... But dunno. Thought I'd check with you first.

Reply:
`marked` one requires more work. Need additional tests too, because I see enough breaking changes which surprisingly aren't breaking anything for us 🤨 

Can you remove `marked` update from this PR? I've https://github.com/w3c/respec/issues/4444 pending specifically for `marked` update.

Reply:
sounds good.